src/v/features/feature_table.cc

```diff
@@ -697,6 +703,40 @@ void feature_table::assert_compatible_version(bool override) {
     }
 }
 
+feature_table::probe::probe(const feature_table& parent)
+  : _parent(parent) {
+    if (ss::this_shard_id() != 0) {
+        return;
+    }
+
+    if (!config::shard_local_cfg().disable_metrics()) {
+        setup_metrics(_metrics);
+    }
+
+    if (!config::shard_local_cfg().disable_public_metrics()) {
+        setup_metrics(_public_metrics);
+    }
```

Comment:
Yeah, that's a fair point. There's no `feature_table::start` method at the moment, and similarly there are no `start` methods on any controller-log-backed `_table.cc`'s. I think that's because they are `ss::sharded<>` and they are expected to be constructed and started at the same time with `ss::sharded::start()`. So what I've done now is I've added a `setup_metrics` method that gets called inside the constructor of `feature_table` right after the probe is created to make it explicit that we start the probe immediately.

Reply:
>expected to be constructed and started at the same time with `ss::sharded::start()`

Not necessarily, you'll see this pattern throughout `application.cc`

https://github.com/redpanda-data/redpanda/blob/dev/src/v/redpanda/application.cc#L2475-L2478

But yeah, it doesn't seem like this probe should need an async lifetime since the lifetime of the `sharded<feature_table>` tracks with the whole program...the test failures[1] are real and maybe a result of feature_table being con/destructed manually across various tests and fixtures. 

[1] [build](https://buildkite.com/redpanda/redpanda/builds/54738#01920aec-ccf8-43d5-828d-1b636d7d9d3b) - these do go away if I set `_probe = nullptr` in `feature_table::feature_table(...)`

Reply:
Those are not `_table`'s though 😉 but yeah fair point, there are certainly examples of decoupled sharded service construction and start. I'm wondering what seastar-people would think of that.

Thanks for the pointer on the build, the cause turned out to be multiple `feature_table`'s being constructed at the same time.